data/org.freedesktop.portal.Notification.xml

```diff
@@ -116,8 +180,212 @@
+
+              Inline replies for instant messaging apps.
+              The user-provided text will be added to the response vardict as ``reply-text`` (``s``);
+
+            Additional purposes may be defined by notification servers using
+            ``x-vendor`` prefix e.g. ``x-gnome.class.specific``
+
+          * ``target`` (``v``)
+
+            Target parameter to send along when activating the action.
+
+            This target will be added to the response vardict as ``target`` (``v``);
+
+
+        * ``desktop-file-id`` (``s``)
+
+          An optional ``desktop-file-id``. Mandatory for un-sandboxed apps;
+
+          If set, this must be a valid
+          `Desktop File ID <https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s02.html#desktop-file-id>`_
+          with ``.desktop`` as a suffix.
+
+          For sandboxed apps, ``desktop-file-id``
+          must start with the app ID and can have a suffix after a period.
+          E.g A sandboxed app with id ``org.libreoffice.LibreOffice``
+          is allowed to specify ``org.libreoffice.LibreOffice.Writer.desktop``.
+
+          The desktop file needs to be installed on the system.
+
+        * ``display-hint`` (``as``)
+
+          An array of ways to display the notification. If none are set, or the notification server has its own policy, it is
+          free to decide how and where to display the notification.
+
+          * ``transient``
+
+            The notification is displayed only as a banner and won't be kept
+            by the server in a tray.
+
+            It's a programmer error to specify ``tray`` at the same time.
+
+          * ``tray``
+
+            No banner for the notification will be displayed and
+            the notification is placed in the tray.
+
+            It's a programmer error to specify ``transient`` at the same time.
+
+          * ``persistent``
+
+            Make the notification persistent in the notification tray.
+            The user can’t dismiss it using the usual close button or gesture.
+
+            Apps are only allowed to display persistent notifications
+            as long as they have a window. Once the last window of an app
+            is closed the persistent notification will be removed.
+
+          * ``hide-on-lockscreen``
+
+            Don't show the notification on the lockscreen.
+
+          * ``hide-content-on-lockscreen``
+
+            All content of the notification will be hidden on the lockscreen.
+
+          * ``show-as-new``
+
+            If a notification with the same ``id`` of the app exists already
+            replace the previous notification, by removing the old notification
+            (including animation, etc) and adding a new notification.
+
+            If this hint isn't specified the notification's content is updated
+            without any flickering.
+
+          Additional purposes may be defined by notification servers.
+
+        * ``content-type`` (``s``)
```

Comment:
I don’t mean to bikeshed, but ‘content type’ pretty much universally means a MIME type (e.g. `application/xml`), so this terminology could be confusing.

Could it be `content-id` or `content-class` or something like that instead?

Reply:
Valid concern. I like `content-class` but we could also stick to `category` like [XDG notification api.](https://specifications.freedesktop.org/notification-spec/notification-spec-latest.html#categories), even though i don't like that much.

Reply:
Ah, if the XDG notification spec already defines something like this, then I think it would be valuable to use the same terminology, i.e. ‘category’. Otherwise people have to re-learn concepts, or use two different names for the same thing, when implementing a notification client or server.